avm/ptn/sa/content-processing/main.bicep

```diff
@@ -0,0 +1,2183 @@
+      location: resourceGroupLocation
+    }
+    managedIdentities: { systemAssigned: true }
+    appLogsConfiguration: {
+      destination: 'log-analytics'
+      logAnalyticsConfiguration: {
+        customerId: logAnalyticsWorkspace.outputs.logAnalyticsWorkspaceId
+        sharedKey: logAnalyticsWorkspace.outputs.primarySharedKey
+      }
+    }
+    workloadProfiles: [
+      {
+        name: 'Consumption'
+        workloadProfileType: 'Consumption'
+      }
+    ]
+    enableTelemetry: enableTelemetry
+    publicNetworkAccess: 'Enabled' // Always enabled for Container Apps Environment
+
+    // <========== WAF related parameters
+
+    platformReservedCidr: '172.17.17.0/24'
+    platformReservedDnsIP: '172.17.17.17'
+    zoneRedundant: (enablePrivateNetworking) ? true : false // Enable zone redundancy if private networking is enabled
+    infrastructureSubnetResourceId: (enablePrivateNetworking)
+      ? avmVirtualNetwork.outputs.subnetResourceIds[1] // Use the container app subnet
+      : null // Use the container app subnet
+  }
+}
+
+// //=========== Managed Identity for Container Registry ========== //
+module avmContainerRegistryReader 'br/public:avm/res/managed-identity/user-assigned-identity:0.4.1' = {
+  name: format(resourceNameFormatString, 'acr-reader-mid-')
+  params: {
+    name: 'acr-reader-mid${solutionPrefix}'
+    location: resourceGroupLocation
+    tags: tags
+    enableTelemetry: enableTelemetry
+  }
+  scope: resourceGroup(resourceGroup().name)
+}
+
+// module bicepAcrPullRoleAssignment 'br/public:avm/ptn/authorization/resource-role-assignment:0.1.2' = {
+//   name: format(resourceNameFormatString, 'rabc-acr-pull')
+//   params: {
+//     resourceId: avmContainerRegistry.outputs.resourceId
+//     principalId: avmContainerRegistryReader.outputs.principalId
+//     roleDefinitionId: '7f951dda-4ed3-4680-a7ca-43fe172d538d' // AcrPull role
+//     principalType: 'ServicePrincipal'
+//     enableTelemetry: enableTelemetry
+//   }
+//   scope: resourceGroup(resourceGroup().name)
+// }
+
+// ========== Container App  ========== //
+module avmContainerApp 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: format(resourceNameFormatString, 'caapp-')
+  params: {
+    name: 'ca-${solutionPrefix}-app'
+    location: resourceGroupLocation
+    environmentResourceId: avmContainerAppEnv.outputs.resourceId
+    workloadProfileName: 'Consumption'
+    enableTelemetry: enableTelemetry
+    registries: useLocalBuild == 'localbuild'
+      ? [
+          {
+            server: publicContainerImageEndpoint
+            identity: avmContainerRegistryReader.outputs.principalId
+          }
+        ]
+      : null
+
+    managedIdentities: {
+      systemAssigned: true
+      userAssignedResourceIds: [
+        avmContainerRegistryReader.outputs.resourceId
+      ]
+    }
+
+    containers: [
+      {
+        name: 'ca-${solutionPrefix}'
+        image: '${publicContainerImageEndpoint}/contentprocessor:latest'
```

Comment:
Hardcoding latest image is not a good idea, specially if at some point, there is a non backward compatible change that depends on the infrastructure (e.g. depends of a new environment variable).

Recommended to create a parameter with a default value set to a specific version of the image that works for the current - e.g. `param backendContainerImageTag string = 'latest_2025-06-17_704'`

See how we are addressing in [Multi Agent](https://github.com/Azure/bicep-registry-modules/blob/main/avm/ptn/sa/multi-agent-custom-automation-engine/main.bicep#L52)